flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraphBuilder.java

```diff
@@ -296,7 +296,8 @@ public static ExecutionGraph buildGraph(
 				checkpointIdCounter,
 				completedCheckpoints,
 				rootBackend,
-				checkpointStatsTracker);
+				checkpointStatsTracker,
+				chkConfig.getTolerableCheckpointFailureNumber());
```

Comment:
Nit: this method is already taking all other fields from `chkConfig`. It seems like this has super-high cohesion so maybe you could change it to pass in the `chkConfig` instead of all fields on by one. Or at least position this further up with the other calls on `chkConfig`.

Reply:
Agree